edge-modules/api-proxy-module/src/monitors/certs_monitor.rs

```diff
@@ -0,0 +1,305 @@
+use super::utils;
+use anyhow::Result;
+use chrono::Utc;
+use futures_util::future::{self, Either};
+use std::sync::Arc;
+use tokio::sync::Notify;
+
+const PROXY_SERVER_TRUSTED_CA_PATH: &str = "/app/trustedCA.crt";
+const PROXY_SERVER_CERT_PATH: &str = "/app/server.crt";
+const PROXY_SERVER_PRIVATE_KEY_PATH: &str = "/app/private_key.pem";
+
+const PROXY_SERVER_CERT_VALIDITY_DAYS: i64 = 90;
+//Time in the past, to trigger an certificate expiration event so the system go pick up certificate
+const EXPIRY_TIME_START_DATE: &str = "1996-12-19T00:00:00+00:00";
+
+//Check for expiry of certificates. If certificates are expired: rotate.
+pub async fn start(notify_certs_rotated: Arc<Notify>, shutdown_signal: Arc<Notify>) {
+    const NGINX_CERTIFICATE_MONITOR_POLL_INTERVAL_SECS: std::time::Duration = std::time::Duration::from_secs(1);
+
+    let module_id = std::env::var("IOTEDGE_MODULEID")
+        .expect(&format!("Missing env var {:?}", "IOTEDGE_MODULEID"));
```

Comment:
Can I discuss that with you? I am following the practice designed here: https://nick.groenen.me/posts/rust-error-handling/#application-errors.
Since this is an application and not a library, I would shoot first for readability.

Reply:
Fixed